lib/Sema/TypeCheckDecl.cpp

```diff
@@ -2756,6 +2756,9 @@ class DeclChecker : public DeclVisitor<DeclChecker> {
     // we're doing here.
     CD->walkSuperclasses(
       [&](ClassDecl *superclass) {
+        if (!superclass->getParentSourceFile())
+          return TypeWalker::Action::Stop;
```

Comment:
The global registration of @objc methods is only used for a few diagnostics about conflicting selectors and such. I think it’s ok if it doesn’t work quite right when you have a mix of source files and SIB.

Reply:
It's also used for AnyObject dispatch, no?

Reply:
No, the table of methods for AnyObject dispatch is separate and built when the first lookup is performed into a module. It can’t be built here - this is typeCheckDecl() so it could happen after some expressions have already been checked, and it won’t run at all for classes in secondary files.